### PR TITLE
devops455 remove instance via API, first commit

### PR DESCRIPTION
**Pull request by francois-vincent**

### CODE REVIEW

File: source/navitiacommon/navitiacommon/models.py [hunk 1]
```diff
@@ -219,6 +219,7 @@ def __init__(self, uri, name=None, instance=None):
 class Instance(db.Model):
     id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.Text, unique=True, nullable=False)
+    discarded = db.Column(db.Boolean, default=False, nullable=False)
```
Comment (kinnou02) [inline source/navitiacommon/navitiacommon/models.py:222]: You need to create a migration file

File: source/tyr/tyr/resources.py [hunk 2]
```diff
@@ -289,18 +289,36 @@ def get(self, id=None, name=None):
...
         args.update({'id': id, 'name': name})
         if any(v is not None for v in args.values()):
-            return models.Instance.query.filter_by(**{k: v for k, v in args.items() if v is not None}).all()
+            return models.Instance.query_existing().filter_by(**{k: v for k, v in args.items() if v is not None}).all()
         else:
-            return models.Instance.query.all()
+            return models.Instance.query_existing().all()
 
-    def put(self, id=None, name=None):
+    def _get_from_id_or_name(self, id, name):
```
Comment (kinnou02) [inline source/tyr/tyr/resources.py]: This should be the responsibility of the model

**Comment by antoine-de** [comment]
I think some tests would be great :eyes:

File: source/tyr/tyr/resources.py (hunk 2 above)
Comment (francois-vincent) [inline source/tyr/tyr/resources.py]: right !

File: source/navitiacommon/navitiacommon/models.py (hunk 1 above)
Comment (francois-vincent) [inline source/navitiacommon/navitiacommon/models.py:222]: done !

File: source/tyr/migrations/versions/d1d12707b76_add_discarded_instance.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+revision = 'd1d12707b76'
+down_revision = '4fa7f3f529e'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.add_column('instance', sa.Column('discarded', sa.Boolean(), nullable=False))
```
Comment (kinnou02) [inline source/tyr/migrations/versions/d1d12707b76_add_discarded_instance.py]: This will fail, you need to set a `server_default` for existing instances

**Comment by antoine-de** [comment]
dupplicates with #1620

**Pull request closed without merging**